http/rest-client-reactive-vanilla/src/main/java/io/quarkus/ts/http/restclient/reactive/RestCallerService.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import java.net.URI;
+
+import jakarta.annotation.PostConstruct;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+
+import io.quarkus.runtime.Startup;
+
+/**
+ * This class is used to reproduce issue https://github.com/quarkusio/quarkus/issues/31024
+ * Also requires {@link UselessRestApi} and {@link VersionHeaderFilter} for this
+ */
+@Startup
+public class RestCallerService {
+
+    @PostConstruct
+    void init() {
+        callRest();
+    }
+
+    void callRest() {
+        RestClientBuilder builder = RestClientBuilder.newBuilder()//
+                .baseUri(URI.create("localhost"));
+
+        // API needs to be created for issue to manifest
+        UselessRestApi api = builder.build(UselessRestApi.class);
```

Comment:
Can you actually call something? Building REST client just to check static registration inside REST client extension feels very much like unit test. Then it belongs to Quarkus project (please feel free to contribute it there).

My preference is to write actual scenarios, as if we already pay a price of building this in JVM / native etc. one Vert.x setup and REST call is almost undetectable but we will have another test coverage of extension code (end-to-end not some isolated test).

Reply:
In this case, it probably belongs to quarkus project. There is no reason to actually use the API, neither to do any HTTP calls to the service. AFAIK only detectable way the issue manifest itself is nit the app logs.

WDYT @rsvoboda is it OK if we scap this and move this test to upstream, or should we have it in our TS?

Reply:
okay, I'll respect whatever you go with.

just to be clear - I think you should just use the client and that's it.

Reply:
We've agreed, that there is no real way to use REST or any other API to detect the failure.